From 346f0710c78e522f88b65fa09d7e3cf8c3597b2f Mon Sep 17 00:00:00 2001
From: saravananm <saravananm@ami.com>
Date: Fri, 17 Jan 2025 17:25:14 +0530
Subject: [PATCH] Fix-For-REKEY-RENEW

Description:
The CSR strings of two different actions are the same.

Tested:
Build taken Successfully and verified.

After Fix:
root@AMIOT-2230d5d8a180:~# busctl call xyz.openbmc_project.Certs.Manager.Server.Https /xyz/openbmc_project/certs/server/https/csr xyz.openbmc_project.Certs.CSR CSR
s "-----BEGIN CERTIFICATE REQUEST-----\nMIIC1zCCAb8CAQAwgZExDzANBgNVBAcMBkR1bHV0aDEQMA4GA1UEAwwHYW1pLmNv\nbTELMAkGA1UEBhMCVVMxHjAcBgkqhkiG9w0BCQEWD21lZ2FyYWNAYW1pLmNvbTEN\nMAsGBCsOAwIMA1JTQTEMMAoGA1UECgwDQU1JMRAwDgYDVQQLDAdNRUdBUkFDMRAw\nDgYDVQQIDAdHZW9yZ2lhMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA\n3IV/9EaTOXvI33T3WLIZUqtMRSNqhXuZHquly2RaemH13EFFZLpIXQDLHT57uzyz\nP5O3lustAI/KoRoL+4LsJF7F7YMQIXYETi1tdvdNOyoR4AvBc+cm9MKQyp2cpcqk\nBRd0iCGtR7LNrJCoX58pf43uo3OI9sVf/wanbXoWqicxC6LU+B5lXNwgyqL3bbOP\n85V2/KvkWj0NMDZB/wLW4y5y4kxlRAZChpWSiR7Adp6Gr0P0S4SLf+QwYaW9gflO\nR1eHFtNez0mKmbpBYkoFtjbu7aYEzKUbQOjQ1IX20t7Ec/gE70JknwaO2DUMnvZX\nT+vdnXlBaN5/V2FYH1E5cwIDAQABoAAwDQYJKoZIhvcNAQELBQADggEBAEYpeuW4\nhJ3RtE29uJNiymzGdOYeGwEAz9nOAaEtyNHybgOsBAi/JwtssDxUoGFKhtIk4FTR\nMPr0lWMp3S5EHyzVxvnrdAFLiCJ9VPltkIq4M53sQ+rNN8cNyfkDQgIZGg/BocK/\njfA7mglD93x6BL4EMViOG5L3VuYmaYRx1PAo0ZPcKA4UiJhslaH7qi2Fg+Kinr7l\nXFxb+vgLT4iSkp0xq7QNjXSpaQEsb5Y8Arfj7pMHrHfUg9lwxO5SJyvcVWNvQNEl\nglvc1JH+Iej1m8af1b17BJdHJxbgwWFq4ai1vE1ZXYK+Jv7zP+GUJObMPnTIa6+j\n+Fd6NdPPzcRUNCo=\n-----END CERTIFICATE REQUEST-----\n"
root@AMIOT-2230d5d8a180:~# busctl call xyz.openbmc_project.Certs.Manager.Server.Https /xyz/openbmc_project/certs/server/https xyz.openbmc_project.Certs.CSR.Create reNew
s "/xyz/openbmc_project/certs/server/https/csr
root@AMIOT-2230d5d8a180:~# busctl call xyz.openbmc_project.Certs.Manager.Server.Https /xyz/openbmc_project/certs/server/https/csr xyz.openbmc_project.Certs.CSR CSR
s "-----BEGIN CERTIFICATE REQUEST-----\nMIIC1zCCAb8CAQAwgZExDzANBgNVBAcMBkR1bHV0aDEQMA4GA1UEAwwHYW1pLmNv\nbTELMAkGA1UEBhMCVVMxHjAcBgkqhkiG9w0BCQEWD21lZ2FyYWNAYW1pLmNvbTEN\nMAsGBCsOAwIMA1JTQTEMMAoGA1UECgwDQU1JMRAwDgYDVQQLDAdNRUdBUkFDMRAw\nDgYDVQQIDAdHZW9yZ2lhMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA\nvVbfn/k5hokx5RFNPwluvnYS3EpuKK6maYqN3hObeE+7T5bY9exGXEwcYI0HrZ5F\nQr/oXKH1jj84oCclRFNOgYfS5fhbz+CZqHGTtR7LWfyZA8SyaSaHqnFroAtSop9z\nSwtz36jE9Q3M4kk03jBkx9wdOlNXXMyx+asuwLO8LmAjrq3jmzNO4I3qhrqBUuXQ\nU6xmytDnRlIOxkmmw+DsQnJMVtbhfAWVm+Uq5skHw4DohNrc3jtxeElPrypqNZgx\nvrmsrkwkKWF0fShdKJIu1SfWG6X7CkTmehlF4OAoxaqEpMG/rd1qxPz2vhPBs/0i\npOY9yjUNn3WyZHpqAIPfGwIDAQABoAAwDQYJKoZIhvcNAQELBQADggEBAD+xA0Is\nGEDtrHqNAA78IVhfrsr5aMKxwFRW8x/bhZ3a2CEsqp8qh/6utKDP2dunrC8x0ENb\nQpbfzjcKnt9a1ClvyYSeQljxrUTE691JxhMMOZLqit7CNbIq85Y7DW2KQmbVmtY5\nAVZpxlYB8ZJt01dt9QJbA+N28QbSmedatQBKEPDifFTiDSVREj+JGfUU9I49rkSp\nNJfpH91Uc609Uvi+mnyQlulHRynJe43mzuTPfguBavpPOGLj6QpqcEu7owkISyCC\ncPaXj6st9OlAZ7lY5t6NFakNlg4GD2Z7YlA3BWuGySGf3GQkbAbOuj3GbjtzE9Wh\nzwIf8GlFq4/0UJw=\n-----END CERTIFICATE REQUEST-----\n"

Signed-off-by: saravananm <saravananm@ami.com>
---
 certs_manager.cpp | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/certs_manager.cpp b/certs_manager.cpp
index 380f75b..10d23c4 100644
--- a/certs_manager.cpp
+++ b/certs_manager.cpp
@@ -644,16 +644,7 @@ void Manager::generateCSRHelper(
 {
     int ret = 0;
 
-    // set version of x509 req
-    int nVersion = 1;
     X509ReqPtr x509Req(X509_REQ_new(), ::X509_REQ_free);
-    ret = X509_REQ_set_version(x509Req.get(), nVersion);
-    if (ret == 0)
-    {
-        lg2::error("Error occurred during X509_REQ_set_version call");
-        ERR_print_errors_fp(stderr);
-        elog<InternalFailure>();
-    }
 
     // set subject of x509 req
     X509_NAME* x509Name = X509_REQ_get_subject_name(x509Req.get());
@@ -705,7 +696,7 @@ void Manager::generateCSRHelper(
       }
     // Used EC algorithm as default if user did not give algorithm type.
     if (keyPairAlgorithm == "RSA")
-        pKey = getRSAKeyPair(keyBitLength);
+        pKey = generateRSAKeyPair(keyBitLength);
     else if ((keyPairAlgorithm == "EC") || (keyPairAlgorithm.empty()))
         pKey = generateECKeyPair(keyCurveId);
     else
-- 
2.34.1

